proxy/apifilterproxy/filters/auth/rancher_token_validate_filter.go

```diff
@@ -0,0 +1,289 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/filters"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/model"
+)
+
+const (
+	name = "authTokenValidator"
+)
+
+//AuthorizeData is for the JSON output
+type AuthorizeData struct {
+	Message string `json:"message,omitempty"`
+}
+
+//MessageData is for the JSON output
+type MessageData struct {
+	Data []interface{} `json:"data,omitempty"`
+}
+
+func init() {
```

Comment:
We wanted to have a "pluggable" filter model where people can write against the api_filter interface and just drop in the go file.
That is why we are using this model (similar to the external-dns/external-lb pluggable frameworks)